code/game/machinery/dehumidifier.dm

```diff
@@ -0,0 +1,123 @@
+//This magic dehumidifier is a portable water-to-nothing machine, although it does need a charged cell to work.
+//See drain.dm!
+/obj/machinery/dehumidifier
+	name = "emergency dehumidifier"
+	desc = "A portable dehumidifier with emergency high-volume pump to help keep you dry when it's getting damp... just very slowly."
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "dhum-off"
+	anchored = FALSE
+	density = TRUE
+	use_power = POWER_USE_OFF
+	var/obj/item/cell/cell = /obj/item/cell/high
+	var/on = FALSE
+	var/drainage = TRUE
+	var/last_gurgle = 0
+	atom_flags = ATOM_FLAG_CLIMBABLE
+	clicksound = "switch"
+
+/obj/machinery/dehumidifier/Initialize()
+	if(ispath(cell))
+		cell = new cell(src)
+	. = ..()
+	update_icon()
+
+/obj/machinery/dehumidifier/on_update_icon(var/rebuild_overlay = 0)
+	icon_state = "dhum-[on ? "on" : "off"]"
+	if(rebuild_overlay)
+		overlays.Cut()
+		if(panel_open)
+			overlays.Add("dhum-open")
+
+/obj/machinery/dehumidifier/examine(mob/user)
+	. = ..(user)
+	if (!.)
+		return
```

Comment:
I believe this is an incorrect use of `examine` parent calls; you should do `. = ..()` (no truncating arguments) and then not check the result.